spec/components/find/courses/sumary_component/view_preview.rb

```diff
@@ -51,12 +51,13 @@ def mock_primary_course
                          qualification: 'pgce',
                          funding_type: 'salaried',
                          subjects: nil,
-                         level: :primary)
+                         level: :primary,
+                         can_sponsor_student_visa: true)
         end
 
         class FakeCourse
           include ActiveModel::Model
-          attr_accessor(:provider, :accrediting_provider, :has_bursary, :age_range_in_years, :course_length, :applications_open_from, :start_date, :qualification, :funding_type, :subjects, :level)
+          attr_accessor(:provider, :accrediting_provider, :has_bursary, :age_range_in_years, :course_length, :applications_open_from, :start_date, :qualification, :funding_type, :subjects, :level, :can_sponsor_student_visa)
```

Comment:
Sorry, I'm a bit confused. What do these files do? It's not a spec, are they being used as factories?

Reply:
It's this: https://viewcomponent.org/guide/previews.html

A way to view the components in isolation: https://www.publish-teacher-training-courses.service.gov.uk/view_components/flash_banner/with_error

Reply:
Nice, thanks!